### PR TITLE
Move more functionality from main and Repl into pipeline

### PR DESCRIPTION
**Pull request by HuwCampbell**
Still might try and generalise the types of these. Keeping away from Text and Var for instance.

### CODE REVIEW

File: src/Icicle/Repl.hs [hunk 1]
```diff
@@ -100,8 +109,8 @@ sourceParse = mapLeft ReplErrorCompile . P.sourceParseQT "repl"
 sourceDesugar :: P.QueryTop' -> Either ReplError P.QueryTop'
 sourceDesugar = mapLeft ReplErrorCompile . P.sourceDesugarQT
 
-sourceReify :: P.QueryTop'T -> Either ReplError P.QueryTop'T
-sourceReify = return . P.sourceReifyQT
+sourceReify :: P.QueryTop'T -> P.QueryTop'T
+sourceReify = P.sourceReifyQT
```
Comment (tranma) [inline src/Icicle/Repl.hs]: might as just just export P.sourceReifyQT and remove this?

Comment (HuwCampbell) [inline src/Icicle/Repl.hs]: I did actually have that, but I liked that none of the reexports for the repl said QT at the end, so left it like that.

**Comment by HuwCampbell** [comment]
It's very annoying that we swap between `Var` and `Text`.

File: src/Icicle/Storage/Dictionary/Toml.hs [hunk 2]
```diff
@@ -38,10 +38,10 @@ import           P
 data DictionaryImportError
   = DictionaryErrorIO         E.SomeException
   | DictionaryErrorParsecTOML Parsec.ParseError
-  | DictionaryErrorParsecFunc (P.CompileError Parsec.SourcePos)
+  | DictionaryErrorParsecFunc (P.CompileError Parsec.SourcePos SP.Variable ())
   | DictionaryErrorParse      [DictionaryValidationError]
-  | DictionaryErrorCheck      (P.CompileError Parsec.SourcePos)
-  | DictionaryErrorTransform  (P.CompileError Parsec.SourcePos)
+  | DictionaryErrorCheck      (P.CompileError Parsec.SourcePos SP.Variable ())
+  | DictionaryErrorTransform  (P.CompileError Parsec.SourcePos SP.Variable ())
```
Comment (tranma) [inline src/Icicle/Storage/Dictionary/Toml.hs]: can these be generalised too...?

Comment (HuwCampbell) [inline src/Icicle/Storage/Dictionary/Toml.hs]: Don't think so.

Comment (HuwCampbell) [inline src/Icicle/Storage/Dictionary/Toml.hs]: At least not meaningfully.

**Comment by amosr** [comment]
remind me - what transforms need to be Text? maybe we can just modify them to work on Var.

**Comment by HuwCampbell** [comment]
Icicle.Avalanche.FromCore.namerText, and inside Pipeline there's a bunch of T.pack

**Comment by amosr** [comment]
We might be better off changing those to work on Var, than swapping between the two for no real reason

File: src/Icicle/Pipeline.hs [hunk 3]
```diff
@@ -81,7 +93,7 @@ annotOfError e
     CompileErrorProgram _
      -> Nothing
 
-instance Pretty a => Pretty (CompileError a) where
+instance (IsString b, Ord b, Pretty a, Pretty b, Show a, Show b, Show c) => Pretty (CompileError a b c) where
```
Comment (amosr) [inline src/Icicle/Pipeline.hs]: the ``IsString`` and ``Ord`` seem strange to me. are they really necessary? I can't think offhand why pretty printing an error would require making a variable, so if it is necessary, add a comment please

Comment (HuwCampbell) [inline src/Icicle/Pipeline.hs]: It contains CheckError, which has suggestions, which calls prettyFunFromStrings.

Comment (amosr) [inline src/Icicle/Pipeline.hs]: ah yeah, forgot about that

File: test/cli/repl/t10-avalanche/expected [hunk 4]
```diff
@@ -6,91 +6,91 @@ ok, loaded test/cli/repl/data.psv, 13 rows
...
-  for_facts (gen$fact : ((Sum Error Int), DateTime)) in new {
-    read [Mutable] [(Sum Error (Buf Int))] acc$conv$22 = acc$conv$22;
-    write acc$conv$22 = Sum_fold# [Error] [Buf Int] [(Sum Error (Buf Int))]
+  init [Mutable] [Int] [32macc[0m$[32mc[0m$[32mconv[0m$[32m11[0m = 0 : Int;
+  init [Mutable] [(Sum Error (Buf Int))] [32macc[0m$[32mconv[0m$[32m22[0m = Right Buf 3 [] : (Sum Error (Buf Int));
+  load_resumable [Int] [32macc[0m$[32mc[0m$[32mconv[0m$[32m11[0m;
+  load_resumable [(Sum Error (Buf Int))] [32macc[0m$[32mconv[0m$[32m22[0m;
+  for_facts ([32mgen[0m$[32mfact[0m : ((Sum Error Int), DateTime)) in new {
+    read [Mutable] [(Sum Error (Buf Int))] [32macc[0m$[32mconv[0m$[32m22[0m = [32macc[0m$[32mconv[0m$[32m22[0m;
+    write [32macc[0m$[32mconv[0m$[32m22[0m = Sum_fold# [Error] [Buf Int] [(Sum Error (Buf Int))]
```
Comment (jystic) [inline test/cli/repl/t10-avalanche/expected]: Is there any way we can avoid this escape code stuff in the test output? It makes the diffs pretty hard to read

Comment (HuwCampbell) [inline test/cli/repl/t10-avalanche/expected]: Done

**Comment by HuwCampbell** [comment]
I think I'm done here

**Comment by tranma** [comment]
looks good to me, thanks for this. :banana:

**Pull request merged**